### PR TITLE
chore(website): simplify playground logic

### LINKED ISSUE #7630: Repo: Playground code refactor: Playground.tsx state switches
### Suggestion

From https://github.com/typescript-eslint/typescript-eslint/pull/6843#discussion_r1157795745, referring to all the logical ternaries in `Playground.tsx`:

> i have a feeling that i should extract this to separate component, this and/or condition chain is getting complex

https://github.com/typescript-eslint/typescript-eslint/pull/6843/files#diff-c76545bc907e11e58eb71533f91f9887abb2a21dc4d8d2a389317dffbbf5a72c

There are indeed a lot of logical conditions in that file. It'd be nice to streamline the logic a bit to be more readable / straightforward / less complex. Maybe a dedicated component or two? Maybe an enum that gets switched on in render logic?

### PR DESCRIPTION
**Pull request by abrahamguo**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7630
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

I took a stab at simplifying the playground code a bit...open to feedback and further discussion 👍🏻

### CODE REVIEW

File: packages/website/src/components/Playground.tsx [hunk 1]
```diff
@@ -168,57 +155,56 @@ function Playground(): React.JSX.Element {
...
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    astModel.storedTsAST!
+                  }
                   onHoverNode={setSelectedRange}
                   cursorPosition={position}
                 />
-              )) ||
-              (state.showAST && astModel && (
+              ) : (
                 <ASTViewer
```
Comment (armano2) [inline packages/website/src/components/Playground.tsx:189]: that looks ok to me.

-----

i was meaning to rewrite this at some point to use provider, passing events back and forward make all this code so unreadable and requires us to use stuff like this

File: packages/website/src/components/Playground.tsx [hunk 2]
```diff
@@ -168,57 +155,56 @@ function Playground(): React.JSX.Element {
...
-            )) ||
-              (state.showAST === 'types' && astModel?.storedTsAST && (
+            ) : state.showAST && astModel ? (
+              state.showAST === 'types' ? (
                 <TypesDetails
                   typeChecker={astModel.typeChecker}
-                  value={astModel.storedTsAST}
+                  value={
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    astModel.storedTsAST!
```
Comment (armano2) [inline packages/website/src/components/Playground.tsx:183]: this is not correct, when eslint crashes without reporting ast, this value may not be defined,

eg, null pointer, invalid syntax in rule
ref: https://github.com/typescript-eslint/typescript-eslint/pull/6929

Comment (abrahamguo) [reply packages/website/src/components/Playground.tsx]: thanks for the clarification!

File: packages/website/src/components/Playground.tsx [hunk 3]
```diff
@@ -115,46 +105,43 @@ function Playground(): React.JSX.Element {
...
-            tsconfig={state.tsconfig}
-            eslintrc={state.eslintrc}
-            sourceType={state.sourceType}
-            showAST={state.showAST}
             onASTChange={setAstModel}
             onMarkersChange={setMarkers}
             selectedRange={selectedRange}
             onChange={setState}
-            onLoaded={onLoaded}
+            onLoaded={useCallback((ruleNames, tsVersions) => {
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/Playground.tsx:140]: [Refactor] Err, let's not put `use*` hooks inside components. The [React rules of hooks](https://react.dev/reference/rules/rules-of-hooks) get very hard to keep track of when hooks are called inline.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
This looks a bit cleaner to me 🙂 thanks!

I think the root intent behind #7630 was to make the `Playground.tsx` component a bit more split up even than this. But there's not much velocity in this area of code so I don't think it's super important that we keep iterating. This code reduction makes me happy.

Just requesting the one change on not calling the `useCallback` hook inline.

Laila James opening a door and saying "this is nice"

**Review by JoshuaKGoldberg — APPROVED** [review]
🧹 thanks!

**Pull request merged**